### PR TITLE
Disallow slashes in denom

### PR DESCRIPTION
**Pull request by channa-figure**
## Description

Updated regex checks on denom to not allow slashes.

closes: #258 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Wrote unit and integration [tests](https://github.com/provenance-io/provenance/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [x] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [x] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: x/marker/types/denom.go [hunk 1]
```diff
@@ -25,6 +33,7 @@ const maxDenomMetadataDescriptionLength = 200
 //  - All Denomination unit Aliases have the same SI prefix as their Denom (but maybe different forms, e.g. name vs symbol)
 //  - All Denomination unit Exponents are {SI prefix exponent of the Denom} - {SI prefix exponent of the base}.
 func ValidateDenomMetadataBasic(md banktypes.Metadata) error {
+	sdk.SetCoinDenomRegex(SdkCoinDenomRegex)
```
Comment (iramiller) [inline x/marker/types/denom.go:36]: This must be done in app.go not the module or the rest of the modules will reject the changes.

File: x/marker/types/params.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ const (
 	// DefaultMaxTotalSupply is the upper bound to enforce on supply for markers.
 	DefaultMaxTotalSupply = uint64(100000000000)
 	// DefaultUnrestrictedDenomRegex is a regex that denoms created by normal requests must pass.
-	DefaultUnrestrictedDenomRegex = `[a-zA-Z][a-zA-Z0-9/]{2,64}`
+	DefaultUnrestrictedDenomRegex = `[a-zA-Z][a-zA-Z0-9]{2,64}`
```
Comment (iramiller) [inline x/marker/types/params.go:18]: Let’s add the dash, period in here too for our marker default.

**Review by iramiller — CHANGES_REQUESTED** [review]
The sdk base denom regex must be done in app.go so all the bank stuff and other dependencies pick it up.

File: x/marker/types/denom.go (hunk 1 above)
Comment (channa-figure) [reply x/marker/types/denom.go:36]: Thanks, it didn't feel right putting it there.

**Comment by channa-figure** [comment]
@iramiller The regex is now setup in the app.go for the blockchain.  It also needed to be wired into the `cmd/root.go` in order for it to be picked up by the cli.

**Review by iramiller — APPROVED** [review]
LGTM. Checked the regex in both places, that is what I was expecting to see.

**Pull request merged**